lib/engines/content_block_manager/app/lib/content_block_manager/publishable.rb

```diff
@@ -36,7 +36,12 @@ def schedule_with_rollback
       ActiveRecord::Base.transaction do
         content_block_edition = yield
 
-        content_block_edition.schedule!
+        if content_block_edition.scheduled?
```

Comment:
finding all the different possibilities a bit confusing - I guess this is the first time we are actually editing an existing content block rather than just creating a new one 💫 
Would there be an edge case here where we need to check if the content block edition hasn't been published in the mean time? e.g. if the user hadn't refreshed the view page in a while and then clicks to edit...

Reply:
Maybe in this circumstance we should _only_ let the current edition be edited. If a user clicks to edit an edition that isn't the current edition, we redirect to the current one?

Reply:
That might be a bit tricky actually, as we don't record the latest edition ID until the end of the edit journey, we could have an endpoint like:

```
content-block-manager/content-block/$documentId/schedule/edit
```

Which then finds the latest edition and redirects to the `editions/$editionId/workflow/schedule_publishing` URL